### PR TITLE
Refactor/interpreter/reference

### PR DESCRIPTION
**Pull request by yamacir-kit**
## Types of PR

- [ ] New Features
- [ ] Upgrade of existing features
- [ ] Bugfix

### CODE REVIEW

File: openscenario/openscenario_interpreter/include/openscenario_interpreter/name.hpp [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  }
+
+  friend auto operator<<(std::ostream & os, const Prefixed & prefixed_name) -> std::ostream &
+  {
+    if (prefixed_name.absolute) {
+      os << "{root}::";
+    }
+
+    for (const auto & prefix : prefixed_name.prefixes) {
+      os << (prefix.empty() ? "{annonymous}" : prefix.c_str()) << "::";
```
Comment (hakuturu583) [inline openscenario/openscenario_interpreter/include/openscenario_interpreter/name.hpp:98]: Is it "anonymous"?

File: openscenario/openscenario_interpreter/include/openscenario_interpreter/scope.hpp [hunk 2]
```diff
@@ -48,37 +47,35 @@ class EnvironmentFrame
...
-  /*  */ auto lookupChildScope(const std::string &) const -> std::list<const EnvironmentFrame *>;
+  auto isOutermost() const noexcept -> bool;
+
+private:
+  auto frames(const Name &) const -> std::list<const EnvironmentFrame *>;
 
-  static auto lookupQualifiedElement(
-    const EnvironmentFrame *, std::vector<std::string>::iterator,
-    std::vector<std::string>::iterator) -> Object;
+  auto lookdown(const std::string &) const -> Object;
```
Comment (hakuturu583) [inline openscenario/openscenario_interpreter/include/openscenario_interpreter/scope.hpp:63]: I think look up and look down is two words.
So, I suggest to use "lookDown"

**Review by hakuturu583 — CHANGES_REQUESTED** [review]

File: openscenario/openscenario_interpreter/include/openscenario_interpreter/scope.hpp (hunk 2 above)
Comment (yamacir-kit) [reply openscenario/openscenario_interpreter/include/openscenario_interpreter/scope.hpp:63]: There seems to be no problem with just one word.
https://wikidiff.com/lookdown/lookup

File: openscenario/openscenario_interpreter/include/openscenario_interpreter/name.hpp (hunk 1 above)
Comment (yamacir-kit) [reply openscenario/openscenario_interpreter/include/openscenario_interpreter/name.hpp]: 08ea9ae fixed this.

File: openscenario/openscenario_interpreter/include/openscenario_interpreter/scope.hpp (hunk 2 above)
Comment (kmiya) [reply openscenario/openscenario_interpreter/include/openscenario_interpreter/scope.hpp:63]: FYI: On Wikipedia, Lookdown seems to be used as a fish name (シロガネアジ in Japanese).
<https://en.wiktionary.org/wiki/File:Selene_vomer_03_by_Line1.JPG>

Could not find the word in the other famous dictionaries such as [Cambridge](https://dictionary.cambridge.org/spellcheck/english/?q=lookdown) and [Oxford](https://www.oxfordlearnersdictionaries.com/spellcheck/english/?q=lookdown).

**Review by hakuturu583 — APPROVED** [review]

**Review by hakuturu583 — CHANGES_REQUESTED** [review]
Sorry, still this problem remains.
https://github.com/tier4/scenario_simulator_v2/pull/605#discussion_r747903348

**Review by hakuturu583 — APPROVED** [review]

**Pull request merged**